core/src/main/java/io/questdb/cairo/TableNameRegistryFileStore.java

```diff
@@ -411,7 +381,7 @@ private void writeEntry(TableToken tableToken, int operation) {
         tableNameMemory.putStr(tableToken.getTableName());
         tableNameMemory.putStr(tableToken.getDirName());
         tableNameMemory.putInt(tableToken.getTableId());
-        tableNameMemory.putInt(tableToken.isWal() ? TableNameRegistryFileStore.TABLE_TYPE_WAL : TABLE_TYPE_NON_WAL);
+        tableNameMemory.putInt(tableToken.isWal() ? TableUtils.TABLE_TYPE_WAL : TableUtils.TABLE_TYPE_NON_WAL);
```

Comment:
What if the table is converted from WAL to non-WAL? Here in `tables.d` file will still be a record that it is WAL table.

Reply:
i was checking that file during testing and it is always empty.
it looked to me that tables.d only used if the table is dropped.

Reply:
fixed in [4d2ea43](https://github.com/questdb/questdb/pull/2947/commits/4d2ea4373a21b7d9a2a34593f48388787e451198)